PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/networking/modelcontrollers/Requester.java

```diff
@@ -82,10 +83,31 @@ public abstract class Requester {
 
     public abstract void startAdRequest();
 
+    /**
+     * This Async task is needed because the networkTask needs to be cancelled / destroyed
+     * on a background thread or it throws a NetworkOnMainThreadException
+     **/
+    private static class DestroyNetworkTaskAsyncTask extends  AsyncTask<BaseNetworkTask, Void, Void> {
```

Comment:
I meant that `DestroyNetworkTaskAsyncTask` doesn't have much sense since it just helps to make a call to the thread not from the main thread, but still, it is not correct interprocess communication that may lead to unexpected behavior. 

The `networkTask.cancel(true)` should did all the job.

Reply:
Can we try to remove the DestroyNetworkTaskAsyncTask at all? As for me, it just adds one more multithread call but doesn't influence the general behavior. networks.cancel(true) should stop the thread gracefully without direct communication with the network layer so the error should not appear. 

The error in the initial ticket appears due to the networkTask.destroy(); call wich triggers the network communication on the main thread.